### PR TITLE
modemmanager: update proto handler to include timeout

### PR DESCRIPTION
**Pull request by nickberry17**
Implement a timeout as in qmi.sh to fix https://github.com/openwrt/packages/issues/10829

### CODE REVIEW

**Comment by feckert** [comment]
LGTM ready to merge?

**Comment by BKPepe** [comment]
@feckert no, it needs to bump `PKG_RELEASE`. There are no sights of compile and run tested.

**Comment by feckert** [comment]
@BKPepe Yeah, you're absolutely right. we need to update at least the PKG_Release. Compile test is not necessary this are only script changes.

**Comment by nickberry17** [comment]
> LGTM ready to merge?

No, needs more work

**Comment by Rixerx** [comment]
Can this be cherry picked to 19.07 branch?

"WARNING: Makefile 'package/feeds/luci/luci-proto-modemmanager/Makefile' has a dependency on 'modemmanager', which does not exist"

File: net/modemmanager/files/modemmanager.proto [hunk 1]
```diff
@@ -313,13 +313,15 @@ proto_modemmanager_init_config() {
...
+	proto_config_add_int timeout
 	proto_config_add_defaults
 }
 
 proto_modemmanager_setup() {
 	local interface="$1"
 
-	local modempath modemstatus bearercount bearerpath connectargs bearerstatus beareriface
+	local modempath modemstatus bearercount bearerpath connectargs bearerstatus beareriface timeout
+	[ "$timeout" = "" ] && timeout="10"
```
Comment (aleksander0m) [inline net/modemmanager/files/modemmanager.proto]: I assume you want the "timeout" setting to be part of the modem settings in `/etc/config/network`? if so, this check to intiialize timeout to 10 if not already set should probably be done after having run `json_get_vars` below.

File: net/modemmanager/files/modemmanager.proto [hunk 2]
```diff
@@ -342,6 +344,21 @@ proto_modemmanager_setup() {
 		return 1
 	}
 
+	echo "Waiting for Modem initialization"
```
Comment (aleksander0m) [inline net/modemmanager/files/modemmanager.proto]: Are we truly sure this is right? If netifd attempts to setup the interface and ModemManager is not yet running. the calls below grabbing modemstatus and modempath will end up failing, and the `DEVICE_NOT_MANAGED` error will happen. 

Then, as soon as MM starts, the hotplug scripts will notify that the interface is up and netifd will set it again.

Comment (aleksander0m) [reply net/modemmanager/files/modemmanager.proto]: Will also comment in the original bugreport

**Comment by feckert** [comment]
This pullrequest is pending. So do we need this @nickberry17 or could this get closed.
As described by @aleksander0m https://github.com/openwrt/packages/pull/10837#discussion_r366358258 and so I understand it this is not needed anymore and it should work with the hotplug scripts.

**Comment by neheb** [comment]
This needs to be rebased.

**Comment by aleksander0m** [comment]
I truly think this is not needed at all.

**Comment by nickberry17** [comment]
I'll go ahead and close it

**Pull request closed without merging**